### PR TITLE
Server landingpage dirwatcher

### PR DESCRIPTION
**Pull request by elpaso**
Small improvements in server landing page plugin:

- watcher for project directories
- clear cache if env changes
- add test for empty projects
- remove redundant project id set

### CODE REVIEW

File: src/server/services/landingpage/qgslandingpageutils.cpp [hunk 1]
```diff
@@ -24,23 +24,63 @@
...
+  {
+    QgsLandingPageUtils::QGIS_SERVER_PROJECTS_PG_CONNECTIONS = pgConnections;
+    AVAILABLE_PROJECTS.clear();
+  }
+
+  // Scan QGIS_SERVER_PROJECTS_DIRECTORIES
   if ( AVAILABLE_PROJECTS.isEmpty() )
   {
-    for ( const auto &path : QString( qgetenv( "QGIS_SERVER_PROJECTS_DIRECTORIES" ) ).split( QStringLiteral( "||" ) ) )
+    for ( const auto &path : projectDir.split( QStringLiteral( "||" ) ) )
```
Comment (m-kuhn) [inline src/server/services/landingpage/qgslandingpageutils.cpp:76]: Can you take a const copy before iterating here?

File: src/server/services/landingpage/qgslandingpageutils.cpp [hunk 2]
```diff
@@ -64,35 +104,35 @@ QMap<QString, QString> QgsLandingPageUtils::projects( )
...
+  }
 
-    // PG projects
-    const auto storage { QgsApplication::instance()->projectStorageRegistry()->projectStorageFromType( QStringLiteral( "postgresql" ) ) };
-    Q_ASSERT( storage );
-    for ( const auto &connectionString : QString( qgetenv( "QGIS_SERVER_PROJECTS_PG_CONNECTIONS" ) ).split( QStringLiteral( "||" ) ) )
+  // PG projects (there is no watcher for PG: scan every time)
+  const auto storage { QgsApplication::instance()->projectStorageRegistry()->projectStorageFromType( QStringLiteral( "postgresql" ) ) };
+  Q_ASSERT( storage );
+  for ( const auto &connectionString : QString( qgetenv( "QGIS_SERVER_PROJECTS_PG_CONNECTIONS" ) ).split( QStringLiteral( "||" ) ) )
```
Comment (m-kuhn) [inline src/server/services/landingpage/qgslandingpageutils.cpp:112]: same

File: src/server/services/landingpage/qgslandingpageutils.cpp [hunk 3]
```diff
@@ -24,23 +24,63 @@
 #include "qgsvectorlayer.h"
 #include "nlohmann/json.hpp"
 
+#include <mutex>
```
Comment (m-kuhn) [inline src/server/services/landingpage/qgslandingpageutils.cpp:27]: Required?

File: src/server/services/landingpage/qgslandingpageutils.h [hunk 4]
```diff
@@ -78,6 +78,16 @@ struct QgsLandingPageUtils
    */
   static QString projectUriFromUrl( const QString &url );
 
+  /**
+   * Store env configuration parameter, used to invalidate the cache if the value changes
+   */
+  static QString QGIS_SERVER_PROJECTS_DIRECTORIES;
```
Comment (m-kuhn) [inline src/server/services/landingpage/qgslandingpageutils.h:84]: Can these be made `Q_GLOBAL_STATIC` instead? https://github.com/qgis/QGIS/pull/32552

**Comment by m-kuhn** [comment]
Nice functionality!

File: src/server/services/landingpage/qgslandingpageutils.cpp (hunk 3 above)
Comment (elpaso) [reply src/server/services/landingpage/qgslandingpageutils.cpp:27]: seems so: https://travis-ci.org/github/qgis/QGIS/builds/718542754#L581

File: src/server/services/landingpage/qgslandingpageutils.h (hunk 4 above)
Comment (elpaso) [reply src/server/services/landingpage/qgslandingpageutils.h:84]: I think I'll make them local static instead...

**Review by m-kuhn — APPROVED** [review]
